src/main.cpp

```diff
@@ -18,44 +19,24 @@
 #include <WiFi.h>
 #include "conf.h"
 
-MFRC522DriverPinSimple ss_pin(pins::mfrc522::ss_pin); // Configurable, see typical pin layout above.
-
-MFRC522DriverSPI driver{ss_pin}; // Create SPI driver.
-// MFRC522DriverI2C driver{}; // Create I2C driver.
-MFRC522 mfrc522{driver}; // Create MFRC522 instance.
-
-LCDWrapper<conf::lcd::COLS, conf::lcd::ROWS> LCD(pins::lcd::rs_pin, pins::lcd::en_pin, pins::lcd::d0_pin, pins::lcd::d1_pin, pins::lcd::d2_pin, pins::lcd::d3_pin);
-
-FabServer server(secrets::machine_data::whitelist, secrets::wifi::ssid, secrets::wifi::password);
-
-Machine machine(secrets::machine_data::machine_id, Machine::PRINTER3D, pins::relay::ch1_pin, false);
-
-FabMember candidate_user = FabMember();
-
 static bool ready_for_a_new_card = true;
-
-
+static BoardState board;
 
 void setup()
 {
-  Serial.println("Starting!");
-
   Serial.begin(115200); // Initialize serial communications with the PC for debugging.
-  LCD.begin();
-
-  mfrc522.PCD_Init(); // Init MFRC522 board.
+  Serial.println("Starting setup!");
+  board.init();
 
   Serial.println(F("Scan PICC to see UID, SAK, type, and data blocks..."));
-
+  
   // connection to wifi
-  LCD.state(LCDState::CONNECTING);
-  LCD.update(server, candidate_user, machine);
+  board.changeStatus(BoardStatus::CONNECTING);
   
-  LCD.showConnection(true);
-  LCD.setConnectionState(server.isOnline());
+  board.getLCD().showConnection(true);
+  board.getLCD().setConnectionState(board.getServer().isOnline());
```

Comment:
you probably want to have this in a function called from the loop in a timeout?

- if you showing offline, every tot attempt to reconnect
- and if you showing online, you check you are really online

Reply:
yes, the logic to periodically test the server connection, and report status, is missing. next step.